common/src/main/java/io/druid/metadata/PasswordProvider.java

```diff
@@ -34,5 +34,15 @@
 })
 public interface PasswordProvider
 {
+  // Used only for getting metadata storage connector password, present here for backwards compatibility
+  @Deprecated
   public String getPassword();
+
+  /**
+   * Implementations should provide a way to get password for the input key.
+   * For example {@link DefaultPasswordProvider} looks for a property named
+   * {@param key} prefixed with {@link DefaultPasswordProvider#PWD_PROPERTY_PREFIX}
+   * in the runtime properties and return the value.
+   */
+  public String getPassword(String key);
```

Comment:
Hm, if key is regarded as a standardized "named secret" and not as the full configuration for accessing that secret, I think this interface could work.

Maybe the configuration for the custom PasswordProvider implementing the cloud KMS fetch+decrypt could have value mappings for "Druid-TLS-keyStorePassword" and ""Druid-TLS-keyManagerPassword" in a map somewhere, containing the blob storage paths and KMS coordinates for each "named secret"